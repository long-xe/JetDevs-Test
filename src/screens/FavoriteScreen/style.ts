import { StyleSheet } from 'react-native';
import { Colors, Radius } from '../../styles';
import { scale } from '../../utils/scaling';

export const styles = StyleSheet.create({
  container: { flex: 1 },
  cardWrapper: {
    padding: scale(10),
    width: '100%',
    backgroundColor: Colors.white,
    borderBottomWidth: 1,
    borderBottomColor: Colors.stroke,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    gap: scale(10),
  },
  avatarWrapper: {
    borderWidth: 2,
    borderColor: Colors.white,
    borderRadius: 50,
  },
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 50,
  },
  favoriteImg: {
    width: 24,
    height: 24,
    tintColor: Colors.primary,
  },
  cardContent: {
    flexDirection: 'row',
    gap: scale(10),
    alignItems: 'center',
  },
});
