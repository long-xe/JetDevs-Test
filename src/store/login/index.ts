import { reducer } from "./login.slice";

export default {
    reducer,
}