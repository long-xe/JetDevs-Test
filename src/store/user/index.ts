import { reducer } from "./user.slice";
import * as actions from './user.action'

export default {
    reducer,
    ...actions,
}