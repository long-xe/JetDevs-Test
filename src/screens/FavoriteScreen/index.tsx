import { View, FlatList } from 'react-native';
import React from 'react';
import { useSelector } from 'react-redux';
import type { RootState } from '../../store/store';

import { styles } from './style';
import UserCard from '../../components/cards/favorite/userCard';

export default function FavoriteScreen() {
  const { favoriteUser } = useSelector((state: RootState) => state.users);

  return (
    <View style={styles.container}>
      <FlatList
        data={favoriteUser}
        keyExtractor={(item, idx) => `${idx + item.login.uuid}`}
        renderItem={({ item }) => {
          return <UserCard user={item} />;
        }}
      />
    </View>
  );
}
