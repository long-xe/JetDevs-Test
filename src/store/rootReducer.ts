
import login from './login';
import user from './user';

export const rootReducer = {
  users: user.reducer,
  login: login.reducer
};
