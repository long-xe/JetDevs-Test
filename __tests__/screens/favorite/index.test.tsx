import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { act } from 'react-test-renderer';
import { store } from '../../../src/store/store';
import FavoriteScreen from '../../../src/screens/FavoriteScreen';


describe('favorite screen', () => {
    it('should renders screen correctly', () => {
      const page = render(
          <Provider store={store}>
            <FavoriteScreen />
          </Provider>
      );
      expect(page).toMatchSnapshot();
    });
  });