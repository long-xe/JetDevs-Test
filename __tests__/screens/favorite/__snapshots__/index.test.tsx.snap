// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorite screen should renders screen correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
